apps/meteor/client/views/omnichannel/departments/DepartmentsRoute.js

```diff
@@ -44,11 +44,16 @@ function DepartmentsRoute() {
 	const debouncedParams = useDebouncedValue(params, 500);
 	const debouncedSort = useDebouncedValue(sort, 500);
 	const onlyMyDepartments = true;
-	const query = useQuery(debouncedParams, debouncedSort, onlyMyDepartments);
 	const departmentsRoute = useRoute('omnichannel-departments');
 	const context = useRouteParameter('context');
 	const id = useRouteParameter('id');
 
+	const refetchRef = useRef(() => undefined);
+
+	const handleRefetch = useCallback(() => {
+		refetchRef.current();
+	}, [refetchRef]);
+
```

Comment:
You must not catch a `QueryResult.refetch` callback from a children component. Instead, invalidate by query key on the query client:

```ts
// ❌  don't do this
const handleRefetch = useCallback(() => {
  refetchRef.current();
}, []);

// 👍 do this
const queryClient = useQueryClient();

const handleRefetch = useCallback(() => {
  const queryClient = [...]; // here goes the child component's query key
  queryClient.invalidateQueries(queryKey);
}, [queryClient]);
```